airflow/models.py

```diff
@@ -3756,8 +3755,7 @@ def sync_to_db(dag, owner, sync_time, session=None):
             DagModel).filter(DagModel.dag_id == dag.dag_id).first()
         if not orm_dag:
             orm_dag = DagModel(dag_id=dag.dag_id)
-            log = LoggingMixin().logger
-            log.info("Creating ORM DAG for %s", dag.dag_id)
+            dag.logger.info("Creating ORM DAG for %s", dag.dag_id)
```

Comment:
This is odd. You are logging in a different class here. I kind of assume a log is private to the class (although we are not explicitly doing that)

Reply:
Not the subject of this PR, but this being a static method, where the first argument is an instance of the class is very odd.

There are two uses. One in utils/db.py:

```
    for dag in dagbag.dags.values():
        models.DAG.sync_to_db(dag, dag.owner, now)
```

This would make more sense as `dag.sync_to_db(synctime=now)`. The other use is inside this method itself:

```
        for subdag in dag.subdags:
            DAG.sync_to_db(subdag, owner, sync_time, session=session)
```

which could be become `subdag.sync_to_db(owner=self.owner, sync_time=sync_time, session=session)`

Think that's worth adding as a standalone PR and not changing this logger use?

Reply:
(or not changing this logger use now, but in that PR)

Reply:
@ashb absolutely, that makes sense. A PR would be very welcome.

Reply:
Raised as #2605 (which will need this PR)